node-manager/src/socket.rs

```diff
@@ -140,17 +141,19 @@ impl MultiWsSocketDescriptor {
         self.connected.load(Ordering::Relaxed)
     }
 
-    pub fn reconnect(&mut self, conn: Arc<Proxy>) {
-        let (send_to_multi_socket, read_from_sub_socket): (Sender<Message>, Receiver<Message>) =
-            unbounded();
-
-        let socket_map: Arc<Mutex<HashMap<Vec<u8>, Sender<Message>>>> =
-            Arc::new(Mutex::new(HashMap::new()));
+    pub async fn reconnect(&mut self, conn: Arc<Proxy>) {
+        debug!("setting up multi websocket descriptor");
+        // if reconnecting master socket, disconnect and clear all subsockets
+        for (_id, (subsocket, _sender)) in self.socket_map.lock().await.iter_mut() {
+            // tell the subsocket to stop processing
+            // and ldk to disconnect that peer
+            subsocket.disconnect_socket();
+            self.peer_manager
+                .socket_disconnected(&WsSocketDescriptor::Mutiny(subsocket.clone()));
+        }
 
+        self.socket_map.lock().await.clear();
```

Comment:
you call `self.socket_map.lock()` twice here, would be better to do `let socket_map = self.socket_map.lock()`, there's probably a potential race here with that